test/integration/targets/win_psmodule/tasks/main.yml

```diff
@@ -1,28 +1,97 @@
+      win_package:
+        path: '{{ temp_path.stdout | trim }}\PackageManagement_x64.msi'
+        state: present
+        creates_path: 'C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\PowerShellGet.psd1'
+
+    - name: create the required folder
+      win_file:
+        path: 'C:\Program Files\PackageManagement\ProviderAssemblies'
+        state: directory
+
+    - name: download nuget
+      win_get_url:
+        url: https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
+        dest: 'C:\Program Files\PackageManagement\ProviderAssemblies\NuGet-anycpu.exe'
+
+    - name: update NuGet provider
+      win_shell: 'Find-PackageProvider -Name Nuget -ForceBootstrap -IncludeDependencies'
+
+    - name: get a random folder path name
+      win_shell: "([System.IO.Path]::GetTempPath()+([System.IO.Path]::GetRandomFileName())).Split('.')[0]"
+      register: random_folder
+
+    - name: create a random folder
+      win_shell: 'New-Item -Path {{ random_folder.stdout | trim }} -ItemType Directory | Out-Null'
+
+    - name: download and save the nevest version of the PackageManagement module from PowerShell Gallery
+      win_shell: 'Save-Module -Name PackageManagement, PowerShellGet -Path {{ random_folder.stdout | trim }} -Force'
+
+    - name: unload PackageManagement and PowerShellGet modules
+      win_shell: 'Remove-Module -Name PackageManagement,PowerShellGet -Force -ErrorAction Ignore'
+      ignore_errors: yes
+
+    - name: get PSModulePath
+      win_shell: "$($Env:PSModulePath -Split ';')[0]"
+      register: psmodulepath
+
+    - name: remove older versions of the PackageManagement and PowerShellGet
+      win_shell: "Get-ChildItem -Path {{ psmodulepath.stdout | trim }} -Directory | Where { @('PackageManagement','PowerShellGet') -contains $_.Name }  | Remove-Item -Recurse -Force -ErrorAction Ignore"
+      ignore_errors: yes
+
+    - name: create required folder
+      win_shell: 'if ( -not $(Test-Path {{ psmodulepath.stdout | trim }}) ) { New-Item -Path {{ psmodulepath.stdout | trim }} -ItemType Directory | Out-Null }'
+
+    - name: update PowerShellGet and PackageManagement modules
+      win_shell: 'Copy-Item -Path {{ random_folder.stdout | trim }}\* -Destination {{ psmodulepath.stdout | trim }}\ -Recurse -Force'
+
+- name: update NuGet version
+  when: powershell_major_version.stdout | int >= 5
+  win_shell: |
+    $nuget_exists = (Get-PackageProvider | Where-Object { $_.Name -eq 'Nuget' } | Measure-Object).Count -eq 1
+
+    if ( $nuget_exists ) {
+      $nuget_outdated = (Get-PackageProvider -Name NuGet -ErrorAction Ignore).Version -lt [Version]"2.8.5.201"
+    }
+
+    if ( -not $nuget_exists -or $nuget_outdated ) {
+      Find-PackageProvider -Name Nuget -ForceBootstrap -IncludeDependencies -Force
+    }
+
+- name: perform cleanup - 1
+  include: clean.yml
+
+- name: run tests
+  include: tests.yml
+
+- name: peform backward compatibility tests
+  when: powershell_major_version.stdout | int == 5
```

Comment:
Why do we have this version check when we've installed PSGet on older versions above?

Reply:
The legacy tests fail at PowerShell 3.0 and I don't want update them. Test cases are covered by the primary set of tests.